src/main/java/hanshyn/onlinebookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+package hanshyn.onlinebookstore.service.cart;
+
+import hanshyn.onlinebookstore.dto.cart.CartItemsRequestDto;
+import hanshyn.onlinebookstore.dto.cart.CartItemsResponseDto;
+import hanshyn.onlinebookstore.dto.cart.CartItemsUpdateRequestDto;
+import hanshyn.onlinebookstore.dto.cart.ShoppingCartDto;
+import hanshyn.onlinebookstore.exception.EntityNotFoundException;
+import hanshyn.onlinebookstore.mapper.CartItemMapper;
+import hanshyn.onlinebookstore.mapper.ShoppingCartMapper;
+import hanshyn.onlinebookstore.model.Book;
+import hanshyn.onlinebookstore.model.CartItem;
+import hanshyn.onlinebookstore.model.ShoppingCart;
+import hanshyn.onlinebookstore.model.User;
+import hanshyn.onlinebookstore.repository.book.BookRepository;
+import hanshyn.onlinebookstore.repository.shopping.CartItemRepository;
+import hanshyn.onlinebookstore.repository.shopping.ShoppingCartRepository;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId()).orElseThrow(
+                () -> new RuntimeException("Can't found user: user")
+        );
+
+        Set<CartItem> cartItem = cartItemRepository.findByShoppingCartId(shoppingCart.getId());
+
+        Set<CartItemsResponseDto> cartItemsResponseDtos = cartItem.stream()
+                .map(cartItemMapper::toDo)
+                .collect(Collectors.toSet());
+
+        ShoppingCartDto shoppingCartDto = shoppingCartMapper.toDto(shoppingCart);
+        shoppingCartDto.setCartItems(cartItemsResponseDtos);
+
+        return shoppingCartDto;
+    }
+
+    @Override
+    public CartItemsResponseDto saveBookToShoppingCart(Authentication authentication,
+                                                       CartItemsRequestDto cartItemsRequestDto) {
+        User user = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseGet(() -> createNewShoppingCart(user));
```

Comment:
Sorry, but I cannot understand logic.
If we try to get ShoppingCart and haven't found it we throw exception. But if we adding item and haven't found cart we create it.
How should get operation work before first insert?

In general: shop is place where we need users in order them to buy something. User without cart has no any value.
For me it's more logical to create cart during registration process and assume that it's exist by default and handle case when cart wasn't found as some data integrity violation